### PR TITLE
Removed Moq dependency from Exporter.OpenTelemetryProtocol.Tests

### PR DESCRIPTION
**Pull request by ngruson**
Related to https://github.com/open-telemetry/opentelemetry-dotnet/issues/4758.
Refactor in order to remove the Moq dependency from the `OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests` project.

## Changes

- Added `MockHttpMessageHandler`
- Added `MockExportClient`
- Removed Moq dependency

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [X] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Mock/MockExportClient.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests.Mock;
+
+internal class MockExportClient<T>(bool throwException = false) : IExportClient<T>
```
Comment (vishweshbankwar) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Mock/MockExportClient.cs:21]: same as https://github.com/open-telemetry/opentelemetry-dotnet/pull/5116#discussion_r1414196880

**Review by vishweshbankwar — APPROVED** [review]

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Implementation/ExportClient/OtlpHttpTraceExportClientTests.cs [hunk 2]
```diff
@@ -94,51 +92,20 @@ public void SendExportRequest_ExportTraceServiceRequest_SendsCorrectHttpRequest(
             Headers = $"{header1.Name}={header1.Value}, {header2.Name} = {header2.Value}",
         };
 
-        var httpHandlerMock = new Mock<HttpMessageHandler>();
+        var httpHandlerMock = new TestHttpMessageHandler();
```
Comment (utpilla) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/Implementation/ExportClient/OtlpHttpTraceExportClientTests.cs:95]:
```suggestion
        var testHttpHandler = new TestHttpMessageHandler();
```

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs [hunk 3]
```diff
@@ -589,37 +587,34 @@ public void CheckToOtlpLogRecordRespectsAttributeLimits()
     public void Export_WhenExportClientIsProvidedInCtor_UsesProvidedExportClient()
     {
         // Arrange.
-        var fakeExportClient = new Mock<IExportClient<OtlpCollector.ExportLogsServiceRequest>>();
+        var fakeExportClient = new TestExportClient<OtlpCollector.ExportLogsServiceRequest>();
```
Comment (utpilla) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs:590]:
```suggestion
        var testExportClient = new TestExportClient<OtlpCollector.ExportLogsServiceRequest>();
```

**Comment by ngruson** [comment]
Test failed due a Docker error 503 Service Unavailable in SqlClient.Tests.
Seems unrelated to this PR?

**Review by utpilla — APPROVED** [review]

**Pull request merged**